public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -125,7 +125,7 @@ export class CloudWatchDatasource
     this.logsTimeout = instanceSettings.jsonData.logsTimeout || '15m';
     this.sqlCompletionItemProvider = new SQLCompletionItemProvider(this, this.templateSrv);
     this.metricMathCompletionItemProvider = new MetricMathCompletionItemProvider(this, this.templateSrv);
-    this.variables = new CloudWatchVariableSupport(this);
+    this.variables = new CloudWatchVariableSupport(this, this.templateSrv);
```

Comment:
It feels like because you already pass it `this` that templateSrv should already be passed in as an argument to CloudWatchVariableSupport, am I misunderstanding?

Reply:
Since `templateSrv` is a private variable, it can't be accessed by `CloudWatchVariableSupport` unless it's passed separately. It was already being passed separately on lines 126 and 127,  so I assumed that it was private for a reason.